src/nupic/algorithms/backtracking_tm_cpp.py

```diff
@@ -154,6 +163,44 @@ def __init__(self,
                             )
 
 
+  @classmethod
+  def getSchema(cls):
+    return BacktrackingTMCppProto
+
+
+  def write(self, proto):
+    super(BacktrackingTMCPP, self).write(proto.baseTM)
+    self.cells4.write(proto.cells4)
+    proto.makeCells4Ephemeral = self.makeCells4Ephemeral
+    proto.seed = self.seed
+    proto.checkSynapseConsistency = self.checkSynapseConsistency
+    proto.initArgs = json.dumps(self._initArgsDict)
+
+
+  @classmethod
+  def read(cls, proto):
+    # Use base class to create initial class
+    obj = BacktrackingTM.read(proto.baseTM)
+    obj.__class__ = cls
```

Comment:
I just didn't have confidence that modifying the class `obj.__class__ = cls` wouldn't cause subtle issues. I will accept this if it's accepted practice in python.